apps/angular/src/app/features/anime-catalog/anime-catalog.component.ts

```diff
@@ -1,17 +1,203 @@
+import { AnimeApiService } from '@js-camp/angular/core/services/anime.service';
+import { Subscription } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { Pagination } from '@js-camp/core/models/pagination.model';
+import { Anime } from '@js-camp/core/models/anime';
+import { CommonModule } from '@angular/common';
+import { animeSelectType, SelectType } from '@js-camp/angular/core/utils/anime-type-select';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Sort } from '@angular/material/sort';
 
 /** A component that represents anime catalog page. */
 @Component({
 	selector: 'camp-anime-catalog',
 	imports: [
 		AnimeTableComponent,
 		HeaderComponent,
+		MatPaginatorModule,
+		MatFormFieldModule,
+		MatSelectModule,
+		MatInputModule,
+		FormsModule,
+		MatIconModule,
+		MatButtonModule,
+		CommonModule,
 	],
 	styleUrl: './anime-catalog.component.css',
 	templateUrl: './anime-catalog.component.html',
-	changeDetection: ChangeDetectionStrategy.OnPush,
 	standalone: true,
 })
-export class AnimeCatalogComponent {}
+export class AnimeCatalogComponent implements OnInit, OnDestroy {
+	/** Child MatPaginator component. */
+	@ViewChild('paginator') private readonly paginator!: MatPaginator;
+
+	/** An array of available anime types to choose from. */
+	protected readonly selectTypes: SelectType[];
+
+	/** A chosen anime type to filter by. */
+	protected selectedType: string | null;
+
+	/** A query to search anime by title. */
+	protected searchQuery: string | null;
+
+	/** A number of anime displayed on a current page. */
+	protected pageSize: number;
+
+	/** An index of the current page. */
+	protected pageIndex: number;
+
+	/** Anime pagination data to be displayed. */
+	protected paginationData?: Pagination<Anime>;
+
+	/** An array of parameters to sort anime by. */
+	protected readonly sortParameters: string[];
+
+	private routeSubscription?: Subscription;
+
+	private readonly route = inject(ActivatedRoute);
+
+	private router = inject(Router);
+
+	private readonly animeApiService = inject(AnimeApiService);
+
+	public constructor() {
+		this.selectTypes = animeSelectType;
+		this.searchQuery = null;
+		this.selectedType = null;
+		this.pageSize = 5;
+		this.pageIndex = 0;
+		this.sortParameters = [];
+	}
+
+	/** Subscribes on route parameters when the component is initialized. */
+	public ngOnInit(): void {
+		this.routeSubscription = this.route.queryParams.pipe(
+			switchMap(params => {
+				if (Object.keys(params).length === 0) {
+					this.router.navigate([], {
+						queryParams: {
+							offset: 0,
+							limit: 5,
+						},
+						queryParamsHandling: 'merge',
+					});
+				}
+				const offset = +params['offset'] || 0;
+				const ordering = params['ordering'] || null;
+				this.searchQuery = params['search'] || null;
+				this.selectedType = params['type'] || null;
+				this.pageSize = +params['limit'] || 5;
+				this.pageIndex = offset / this.pageSize;
+				return this.animeApiService.getPagination(
```

Comment:
I think this is not the best name for the method. I would not specify the detail at all that the anime is patinated, but in extreme cases I would call `getPaginatedAnime`.
```suggestion
				return this.animeApiService.getAll(
```